### PR TITLE
Implement MpiSetDesired and MpiGetReported 

### PR DESCRIPTION
**Pull request by robertschaedler3**
## Description

Implement `MpiSetDesired` and `MpiGetReported` for configuration and reporting via local files. Using the following, files/schemas to set desired objects and 

`/etc/osconfig/osconfig_desired.json`

```json
[
    {
        "component1": {
            "object1": "value1"
        }
    },
    {
        "component1": {
            "object2": "value2"
        },
        "component2": {
            "object3": "value3"
            "object4": {
              "property1": "settingValue1",
              "property2": "settingValue2"
            }
        }
    }
]
```

`/etc/osconfig/osconfig_reported.json`

```json
{
    "component1": {
        "object1": "value1",
        "object2": "value2"
    },
    "component2": {
        "object3": "value3"
        "object4": {
          "property1": "settingValue1",
          "property2": "settingValue2"
        }
    }
}
```

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] I added unit-tests to validate my changes. All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

**Comment by MariusNi** [comment]
The PR descrioption appears to include only simple objects that contain only one setting each. Most MIM objects are complex, they contain multiple settings or different types. Please add such examples and make sure these cases are validated.

**Comment by MariusNi** [comment]
> #define MODULESMANAGERTESTS_H

Avoid double extensions. Try to avoid *.h.in please. 

---
Refers to: src/platform/modulesmanager/tests/ModuleManagerTests/cmake/ModulesManagerTests.h.in:2 in 11ec8c3. [](commit_id = 11ec8c3dce86825f8fd2edb642a89626539c3db6, deletion_comment = False)

File: src/platform/modulesmanager/tests/ModuleManagerTests/src/ModulesManagerTests.cpp [hunk 1]
```diff
@@ -85,6 +102,117 @@ namespace Tests
...
+        constexpr const char payload[] = R""""(
+            [
+                {
+                    "TestComponent1": {
+                        "testParameter": "testValue"
+                    },
+                    "TestComponent2": {
+                        "testParameter": "testValue"
+                    }
+                }
```
Comment (MariusNi) [inline src/platform/modulesmanager/tests/ModuleManagerTests/src/ModulesManagerTests.cpp:130]: This is incorrect multiple componenst case. 

Multiple  components could be CommandRunner and Firewall, for example.

Use real looking MIM components and objects and settings, please.

And add complex objects to the simple ones

File: src/platform/modulesmanager/tests/ModuleManagerTests/src/ModulesManagerTests.cpp [hunk 2]
```diff
@@ -85,6 +102,117 @@ namespace Tests
...
+                    }
+                },
+                {
+                    "TestComponent1": {
+                        "testParameter": "testValue"
+                    },
+                    "TestComponent2": {
+                        "testParameter": "testValue"
+                    }
+                }
```
Comment (MariusNi) [inline src/platform/modulesmanager/tests/ModuleManagerTests/src/ModulesManagerTests.cpp:152]: This is not sufficient

We need: 

- Components with 0, 1 and N objects
- Objects with 0, 1 and N settings
- Settings with either missing, wrong or good type, invalid and valid values

File: src/platform/modulesmanager/tests/ModuleManagerTests/src/MpiTests.cpp [hunk 3]
```diff
@@ -107,4 +107,42 @@ namespace Tests
...
+    {
+        std::string clientName = "MpiTests.MpiSetDesired";
+        char payload[] = R""""(
+            [
+                {
+                    "TestComponent1": {
+                        "testParameter": "testValue"
+                    }
+                }
+            ])"""";
```
Comment (MariusNi) [inline src/platform/modulesmanager/tests/ModuleManagerTests/src/MpiTests.cpp:121]: Insufficient, please see other comments

File: src/platform/modulesmanager/tests/ModuleManagerTests/src/ModulesManagerTests.cpp [hunk 4]
```diff
@@ -85,6 +102,117 @@ namespace Tests
...
     }
 
+    TEST_F(ModuleManagerTests, MpiSetDesiredSingleComponent)
+    {
+        constexpr const char payload[] = R""""(
+            [
+                {
+                    "TestComponent1": {
+                        "testParameter": "testValue"
+                    }
```
[the remark this answers is not in the document]
Comment (robertschaedler3) [reply src/platform/modulesmanager/tests/ModuleManagerTests/src/ModulesManagerTests.cpp:112]: Updated tests to clarify the use of objects vs settings.

File: src/platform/modulesmanager/tests/ModuleManagerTests/src/ModulesManagerTests.cpp (hunk 2 above)
Comment (robertschaedler3) [reply src/platform/modulesmanager/tests/ModuleManagerTests/src/ModulesManagerTests.cpp:158]: Updated tests to clarify the use of objects vs settings.

File: src/platform/modulesmanager/tests/ModuleManagerTests/src/MpiTests.cpp (hunk 3 above)
Comment (robertschaedler3) [reply src/platform/modulesmanager/tests/ModuleManagerTests/src/MpiTests.cpp:121]: Updated tests to clarify the use of objects vs settings.

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**